src/app/article/components/edit-article/edit-article.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Subject, takeUntil } from 'rxjs';
+import { IArticle } from '../../interfaces/article.interface';
+import { ArticlesService } from '../../services/articles.service';
+
+@Component({
+  selector: 'app-edit-article',
+  templateUrl: './edit-article.component.html',
+  styleUrls: ['./edit-article.component.scss']
+})
+export class EditArticleComponent implements OnInit {
+
+  public article: IArticle;
+  public currentArticleId: number;
+  public previewImage: string;
+  public articleForm: FormGroup;
+  private formBuilder = inject(FormBuilder);
+  private articlesSvc = inject(ArticlesService);
+  private activeRoute = inject(ActivatedRoute);
+  private destroy$ = new Subject();
+
+  constructor() { }
+
+  ngOnInit() {
+    this.activeRoute.params.subscribe(params => {
```

Comment:
If you are using `subscribe` that means that this param can change at any time, even after when the component is fully loaded.
In that case the `getArticle` should've been in the subscription callback function, so when the Id changes you load the new article

To indicate that a subscription should work only once you should do `.pipe(take(1))`
In this case, you can use param snapshot which will give you an object, and no need for a subscription,

But for this kind of components where the route param will not change, and the component needs it to render properly, you should have a resolver, which will get the article for the component before navigation, and the component will use that value.

To go further you can have a guard that will check that the id is present in the URL, otherwise it makes no sense to open the component.